### ISSUE TITLE
Run-Al Pipeline with isolation hyperv not working

### ISSUE THREAD

**Issue by Hofschn8** [microsoft/navcontainerhelper#2358]
Hello, 
i try to run following "Run-AlPipeline" : 
![image](https://user-images.githubusercontent.com/39769648/159425522-e8fc06c3-466b-45e9-8f9d-88c8050dc707.png)

But i get following error output 

![image](https://user-images.githubusercontent.com/39769648/159425987-afca956c-a767-4cf0-a50d-28748de314fc.png)

Even if i am overriding the NewBcContainer part in the script, Al Pipeline is still showing, that process isolation is used.....

BcContainerHelper version  is 3.0.3 

Creating a container with container wizard is working fine with following script 
![image](https://user-images.githubusercontent.com/39769648/159425740-c4792a1e-521e-48de-8a37-a790bfa96bc1.png)

Any help?

Thanks
Best regards
David

**Comment by GeWuNAV** [comment]
I guess after the the attached permalink is an `if ($isolation -eq "")` missing. If the expression is true $isolation gets overwritten no matter if it is set or not.

@freddydk If this is on purpose, may you tell me the reason behind it?

https://github.com/microsoft/navcontainerhelper/blob/4719802eb00b1b6babe398dcb27d26929f808915/ContainerHandling/New-NavImage.ps1#L372

Regards
Georg

P.S.: Also for @freddydk We also tried to set the parameters hashtable and passing it to Run-AlPipeline but this gets overwritten in Run-AlPipeline (see below) by an empty hashtable. Are we doing something wrong?

https://github.com/microsoft/navcontainerhelper/blob/4719802eb00b1b6babe398dcb27d26929f808915/AppHandling/Run-AlPipeline.ps1#L747

**Comment by freddydk** [comment]
In your override, you specify -isolation:hyperv - that is wrong - it needs to be -isolation hyperv (like in your second sample)

**Comment by freddydk** [comment]
@GeWuNAV the reason behind that if statement is to default to process isolation with windows 11 or windows server 2022 - where the host and container OS doesn't have to be exactly the same. Also not sure what you mean by passing parameters to run-alpipeline - the way this is done is as specified in this issue (or by setting parameters in the override), he just made a typo with the colon

**Comment by TheDoubleH** [comment]
> @GeWuNAV the reason behind that if statement is to default to process isolation with windows 11 or windows server 2022 - where the host and container OS doesn't have to be exactly the same. 

@freddydk - With the last update, this is, sadly, no longer the case. With Docker Desktop on Win 11 21H2 (22000.556) process isolation no longer works for me.

Docker Desktop 4.6.1 (76265)
Docker Engine v20.10.13

**Comment by Hofschn8** [comment]
@freddydk as @GeWuNAV  already said, even if i correct the typo, hyperv gets overwritten. 
Changing the script, as GEWU told, fixes the issue, and the container gets created with hyperv. 
Process isolation is not working for me as well anymore.

**Comment by GeWuNAV** [comment]
@freddydk We tried a bunch of variants. The attached screenshots all lead to process isolation. Unless you add the if statement I mentioned before to New-NavImage.ps1

The third screenshot is the thing what I meant by passing by parameters hashtable. We debugged that. $parameters contains the isolation until the line `$Parameters = @{}`.

But all in all I get the same error as @Hofschn8 when using process isolation with Win 11 21H2 (22000.556) as @TheDoubleH mentions.

![isolation1](https://user-images.githubusercontent.com/40358071/159644830-ef194594-8995-4281-9dfa-cfc37f31b06b.jpg)
![isolation2](https://user-images.githubusercontent.com/40358071/159644843-2e9bd054-d6f1-4cb5-9536-86b8b21dffb7.jpg)
![isolation3](https://user-images.githubusercontent.com/40358071/159644863-ac52ca72-57fd-4f31-a594-367abb49df21.jpg)
